src/Poll.tsx

```diff
@@ -339,16 +339,25 @@ function Poll<TData = any, TError = any, TQueryParams = { [key: string]: any }>(
   // Compose Contexts to allow for URL nesting
   return (
     <RestfulReactConsumer>
-      {contextProps => (
-        <ContextlessPoll
-          {...contextProps}
-          {...props}
-          requestOptions={{
-            ...contextProps.requestOptions,
-            ...props.requestOptions,
-          }}
-        />
-      )}
+      {contextProps => {
+        const contextRequestOptions =
+          typeof contextProps.requestOptions === "function"
+            ? contextProps.requestOptions()
+            : contextProps.requestOptions || {};
+        const propsRequestOptions =
+          typeof props.requestOptions === "function" ? props.requestOptions() : props.requestOptions || {};
+
+        return (
+          <ContextlessPoll
+            {...contextProps}
+            {...props}
+            requestOptions={{
+              ...contextRequestOptions,
```

Comment:
Almost perfect! But you need to use `merge` from lodash here, so we can add a header without erase these set in the Provider

Example:
https://github.com/contiamo/restful-react/blob/master/src/useGet.tsx#L87

and the associate test:
https://github.com/contiamo/restful-react/blob/eaa37d294e7173a7a1de6a75ac832ab85283a112/src/useGet.test.tsx#L512-L533

Reply:
Right, good catch, thanks!

Fixed and added a test.